<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>793</width>
    <height>518</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>500</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>518</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Maze</string>
  </property>
  <widget class="QWidget" name="widget_central">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="vlayout_screen">
      <item>
       <widget class="QGraphicsView" name="view_screen">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>500</width>
          <height>500</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>500</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">border: none;</string>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="1">
     <layout class="QGridLayout" name="glayout_form">
      <item row="0" column="0">
       <widget class="QTabWidget" name="tabwidget_mazetype">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="tab_maze">
         <attribute name="title">
          <string>Maze</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="2" column="0">
           <widget class="QGroupBox" name="group_generation">
            <property name="title">
             <string>Generation</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="1" column="0">
              <widget class="QLabel" name="label_cols">
               <property name="text">
                <string>cols</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QSpinBox" name="spin_rows">
               <property name="minimum">
                <number>2</number>
               </property>
               <property name="value">
                <number>6</number>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_rows">
               <property name="text">
                <string>rows</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="spin_cols">
               <property name="minimum">
                <number>2</number>
               </property>
               <property name="value">
                <number>6</number>
               </property>
              </widget>
             </item>
             <item row="2" column="0" colspan="2">
              <widget class="QPushButton" name="button_generate">
               <property name="text">
                <string>Generate</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QGroupBox" name="group_file">
            <property name="title">
             <string>File</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_5">
             <item row="0" column="0">
              <widget class="QPushButton" name="button_import">
               <property name="text">
                <string>Import </string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QPushButton" name="button_export">
               <property name="text">
                <string>Export</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QGroupBox" name="group_solution">
            <property name="title">
             <string>Solution</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_4">
             <item row="0" column="0">
              <widget class="QLabel" name="label_x">
               <property name="text">
                <string>X</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item row="1" column="0" colspan="2">
              <widget class="QLabel" name="label_start_point">
               <property name="text">
                <string>Start point</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QSpinBox" name="spin_start_y">
               <property name="minimum">
                <number>1</number>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QSpinBox" name="spin_start_x">
               <property name="minimum">
                <number>1</number>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="label_y">
               <property name="text">
                <string>Y</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item row="5" column="0">
              <widget class="QSpinBox" name="spin_end_x">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="value">
                <number>5</number>
               </property>
              </widget>
             </item>
             <item row="4" column="0" colspan="2">
              <widget class="QLabel" name="label_end_point">
               <property name="text">
                <string>End point</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item row="5" column="1">
              <widget class="QSpinBox" name="spin_end_y">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="value">
                <number>5</number>
               </property>
              </widget>
             </item>
             <item row="6" column="0" colspan="2">
              <widget class="QPushButton" name="button_build_path">
               <property name="text">
                <string>Build path</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_cave">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <attribute name="title">
          <string>Cave</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_10">
          <item row="0" column="0">
           <widget class="QGroupBox" name="group_cave_file">
            <property name="title">
             <string>File</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_7">
             <item row="0" column="0">
              <widget class="QPushButton" name="btn_cave_import">
               <property name="text">
                <string>Import</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QPushButton" name="btn_cave_export">
               <property name="text">
                <string>Export</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QGroupBox" name="group_cave_generation">
            <property name="title">
             <string>Random generation</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_9">
             <item row="3" column="0" colspan="2">
              <widget class="QPushButton" name="btn_cave_generate">
               <property name="text">
                <string>Generate</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="spin_cave_cols">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>50</number>
               </property>
               <property name="value">
                <number>10</number>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QSpinBox" name="spin_cave_rows">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>50</number>
               </property>
               <property name="value">
                <number>10</number>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_cave_cols">
               <property name="text">
                <string>cols</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_cave_rows">
               <property name="text">
                <string>rows</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>probability birth</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QSpinBox" name="spin_cave_probability_birth">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="value">
                <number>50</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QGroupBox" name="group_cave_control">
            <property name="title">
             <string>Control</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_8">
             <item row="0" column="1">
              <widget class="QSpinBox" name="spin_cave_birth_limit">
               <property name="maximum">
                <number>7</number>
               </property>
               <property name="value">
                <number>4</number>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QPushButton" name="btn_cave_auto">
               <property name="text">
                <string>Auto</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_cave_birth_limit">
               <property name="text">
                <string>birth limit</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="spin_cave_death_limit">
               <property name="maximum">
                <number>7</number>
               </property>
               <property name="value">
                <number>3</number>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_cave_death_limit">
               <property name="text">
                <string>death limit</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QPushButton" name="btn_cave_step">
               <property name="text">
                <string>Step</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_cave_interval">
               <property name="text">
                <string>interval (ms)</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QSpinBox" name="spin_cave_interval">
               <property name="maximum">
                <number>9999</number>
               </property>
               <property name="value">
                <number>250</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
